testrpc/testrpc.py

```diff
@@ -83,13 +154,94 @@ def format_block_number(block_number):
 
     return block_number
 
+class LogListener(object):
+    def __init__(self):
+        self.log_idx = 0
+
+    def __call__(self, log):
+        block = evm.block
+        if block.number not in event_log:
+            event_log[block.number] = []
+
+        event_log[block.number].append({
+            "log": log, "log_idx": self.log_idx,
+            "tx_idx": len(block.get_transactions()), "txhash": evm.last_tx.hash,
+            "pending": False, "block": block
+        })
+        self.log_idx += 1
+
+def mine(evm):
+    if evm.block.number == 0:
+        evm.block.log_listeners.append(LogListener())
+
+    listener = evm.block.log_listeners.pop(0)
```

Comment:
Why do you pop out the listener before mining? Doesn't that stop the log events from being collected?